pkg/process/collector.go

```diff
@@ -0,0 +1,121 @@
+package process
+
+import (
+	"bufio"
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs"
+)
+
+type Options struct {
+	Pid            int
+	ProcMountPoint string
+}
+
+type processCollector struct {
+	opts Options
+
+	// Metrics.
+	currMaps *prometheus.Desc
+	maxMaps  *prometheus.Desc
+}
+
+// NewProcessCollector makes a new custom process collector used to collect process metrics the
+// default instrumentation doesn't support.
+func NewProcessCollector(opts Options) prometheus.Collector {
```

Comment:
I'd suggest moving this to linux specific file, so that we can avoid doing the work (opening non-existant files) on other platforms.

For general implementation, NewProcessCollector can return nil, or empty collector.

Reply:
Good suggestion. Will do.

Reply:
Re-thinking it, if I do it, unit tests will not run anymore on OSX. Is this a real issue? Is it a big deal if we try to open a couple of non-existing files on unsupported systems, given Cortex is prod is likely to run on Linux only?

Reply:
It's not a big deal, but it is a wasted work. As an alternative, we can keep the code generic, but check for those files once, and only setup compactor when they exist.

Reply:
Ok. I've added a check to see if the proc FS is supported.